athena-elasticsearch/src/main/java/com/amazonaws/connectors/athena/elasticsearch/ElasticsearchHelper.java

```diff
@@ -0,0 +1,734 @@
+/*-
+ * #%L
+ * athena-example
+ * %%
+ * Copyright (C) 2019 Amazon Web Services
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+package com.amazonaws.connectors.athena.elasticsearch;
+
+import com.amazonaws.athena.connector.lambda.data.Block;
+import com.amazonaws.athena.connector.lambda.data.SchemaBuilder;
+import com.amazonaws.athena.connector.lambda.domain.predicate.EquatableValueSet;
+import com.amazonaws.athena.connector.lambda.domain.predicate.Range;
+import com.amazonaws.athena.connector.lambda.domain.predicate.ValueSet;
+import com.amazonaws.services.elasticsearch.AWSElasticsearch;
+import com.amazonaws.services.elasticsearch.AWSElasticsearchClientBuilder;
+import com.amazonaws.services.elasticsearch.model.DescribeElasticsearchDomainsRequest;
+import com.amazonaws.services.elasticsearch.model.DescribeElasticsearchDomainsResult;
+import com.amazonaws.services.elasticsearch.model.DomainInfo;
+import com.amazonaws.services.elasticsearch.model.ElasticsearchDomainStatus;
+import com.amazonaws.services.elasticsearch.model.ListDomainNamesRequest;
+import com.amazonaws.services.elasticsearch.model.ListDomainNamesResult;
+import com.google.common.base.Splitter;
+import org.apache.arrow.vector.complex.reader.FieldReader;
+import org.apache.arrow.vector.types.Types;
+import org.apache.arrow.vector.types.pojo.Field;
+import org.apache.arrow.vector.types.pojo.FieldType;
+import org.apache.arrow.vector.types.pojo.Schema;
+import org.elasticsearch.common.Strings;
+import org.elasticsearch.index.query.QueryBuilder;
+import org.elasticsearch.index.query.QueryBuilders;
+import org.elasticsearch.search.fetch.subphase.FetchSourceContext;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import java.time.ZoneOffset;
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.time.format.ResolverStyle;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+class ElasticsearchHelper
```

Comment:
add javadoc for the class, what is its purpose, etc...

Reply:
After going through this class I have more feedback here. In general this class feels like it is conflating several responsibilities and should likely be broken up.

1. Do not use statics unless we have a good reason to do so. Nearly all the state in this class should probably not be static and instead be an instance on the appropriate handler (Metadata or Record). The way this is currently written we can not have multiple connectors in the same JVM which will limit how we use this in contexts other than lambda in the future.
2. There is type coercion code in this class in addition to configuration helper methods. Please break out the type stuff into its own class (e.g. TypeUtils.java) as we have done with other connectors.
3. Looks like this class also has tools for building predicates, this should be broken out too. perhaps into a PredicateBuilder.java.

Reply:
break into 3-4 classes.

Reply:
1. Made ElasticsearchHelper a singleton rather than a static class.
2 - 3. Broke up ElasticsearchHelper into 3 additional classes: ElasticsearchQueryUtils, ElasticsearchSchemaUtil, and ElasticsearchTypeUtils. Additionally, moved all field coercion logic to ElasticsearchFieldResolver.